### PR TITLE
Add dev container

### PR DESCRIPTION
**Pull request by ayin21**
# Summary

## Checklist

- [ ] Run `colcon build`
- [ ] Write documentation
- [ ] Create issues for future work
- [ ] Test on your machine
- [ ] Test on the robot
- [ ] This PR is on our `Software` project board

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  && update-locale LANG=en_US.UTF-8 \
+  && ln -s /usr/bin/python3 /usr/bin/python
+
+ENV LANG en_US.UTF-8
+ENV LANGUAGE en_US:en
+ENV LC_ALL en_US.UTF-8
+
+# Setup and prioritize packages.bit-bots.de repository
+RUN apt update -y \
+  && apt upgrade -y --allow-downgrades
```
Comment (jaagut) [inline .devcontainer/Dockerfile:50]:
```suggestion
RUN apt-get update -y \
  && apt-get upgrade -y --allow-downgrades
```

File: .devcontainer/zshrc [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+alias cba='cdc && colcon build --symlink-install --continue-on-error'
+alias cb='cdc && colcon build --symlink-install --continue-on-error --packages-up-to'
+alias cbs='cdc && colcon build --symlink-install --packages-select'
+alias cc='cdc && colcon clean packages --packages-select'
+alias cca='cdc && colcon clean packages'
+
+alias sr='source /opt/ros/iron/setup.zsh && update_ros2_argcomplete'
+alias sc='source $COLCON_WS/install/setup.zsh && update_ros2_argcomplete'
+alias sa='sr && sc'
```
Comment (jaagut) [inline .devcontainer/zshrc:107]: I don't like having these aliases on three places now (Ansible, docs and here). We should just have a `.alias.sh` file or so and source that instead

Comment (Flova) [reply .devcontainer/zshrc:107]: This is a bit hard to implement, as docker build can only access the folder of the Dockerfile.

File: .devcontainer/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    iputils-ping \
+    ipython3 \
+    jq \
+    less \
+    libncurses5-dev \
+    locales \
+    python3-pip \
+    radeontop \
+    ranger \
+    screen \
```
Comment (jaagut) [inline .devcontainer/Dockerfile:24]:
```suggestion
    rsync \
    screen \
```

**Review by Flova — APPROVED** [review]

**Comment by Flova** [comment]
TODO check docs

**Pull request merged**